### ISSUE TITLE
betterDefine doesn't seem to transform `keyof typeof X` correctly.

### ISSUE THREAD

**Issue by nikowonde** [sxzz/unplugin-vue-macros#362]
### Describe the bug

Hi, I've encountered this problem with betterDefine that it doesn't seem to transform `keyof typeof X` correctly while X being an object with key and value pairs being strings.

I've provided quite minimal repro in which Vue throws following warning:
`[Vue warn]: Invalid prop: type check failed for prop "trigger". Expected Null | Array, got String with value "hover".`

### Reproduction

https://stackblitz.com/edit/nuxt-starter-xgkq5y?file=components/Test.vue

### System Info

```Shell
-
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by GODLiangCY** [comment]
I think you may have written the wrong type declaration out of carelessness, the correct way to write it should be
```typescript
const props = defineProps<{
  trigger: keyof typeof Trigger | Array<keyof typeof Trigger>; // works well
}>();
```
instead of
```typescript
const props = defineProps<{
  trigger: keyof typeof Trigger | Array<keyof typeof trigger>;
}>();
```

**Comment by nikowonde** [comment]
> I think you may have written the wrong type declaration out of carelessness, the correct way to write it should be
> 
> ```ts
> const props = defineProps<{
>   trigger: keyof typeof Trigger | Array<keyof typeof Trigger>; // works well
> }>();
> ```
> 
> instead of
> 
> ```ts
> const props = defineProps<{
>   trigger: keyof typeof Trigger | Array<keyof typeof trigger>;
> }>();
> ```

As a matter of a fact, you are correct in a sense that there was a typo with the lowercase t in Trigger in the reproduction. But I am still getting that same error `[Vue warn]: Invalid prop: type check failed for prop "trigger". Expected Null | Array, got String with value "hover".` so I don't think that typo was the root cause of this problem.

**Comment by sxzz** [comment]
Fixed in unplugin-vue-macros@2.1.6

**Comment by nikowonde** [comment]
Great stuff, works like a charm. Thank you very much.
